pages/utils/graph_utils.py

```diff
@@ -1,21 +1,32 @@
 import datetime as dt
 
 
-def get_graph_time_values(interval):
-    # helper values for building graph
+def get_graph_time_values(interval: str | int):
+    """
+    Utility needed in page visualizations- 
+    when the user inputs an 'interval', they're wanting to view
+    data in some time window. This function converts that
+    'interval' value to the necessary Plotly figure update values.
+
+    Args:
+    -----
+        interval (str | int): How long between time bins, user selected.
+
+    Returns:
+    --------
+        x_r ([str]): Time-bin strings.
+        x_name (str): Name of bin-duration.
+        hover (str): String-structure for graph to print on mouse-hover.
+        period (str): How long selected time bin is.
+    """
     today = dt.date.today()
     x_r = None
     x_name = "Year"
     hover = "Year: %{x|%Y}"
     period = "Y"
 
     # graph input values based on date interval selection
-    if interval == 86400000:  # if statement for days
-        x_r = [str(today - dt.timedelta(weeks=4)), str(today)]
-        x_name = "Day"
-        hover = "Day: %{x|%b %d, %Y}"
-        period = "D"
-    elif interval == "D":  # if statement for days
+    if interval == 86400000 or interval == "D":  # if statement for days
```

Comment:
Did you test if this works? When I did it like this it was not producing the right results in all cases

Reply:
haven't tested it, I'll check to see if it seems to comply on my end.

Reply:
@JamesKunstle will mark as resolved once its checked into

Reply:
I don't see any problems by doing this, the graphs look the same across the two implementations.

Reply:
once the changes for the piplock is in ill test it out on my side too just to make sure. Just remember this being a niche issue and want to make 100% sure everything is good